alf/algorithms/predictive_representation_learner.py

```diff
@@ -347,9 +386,13 @@ def preprocess_experience(self, experience: Experience):
             # [B, T, unroll_steps+1]
             positions = torch.min(positions, episode_end_positions)
 
-            # [B, T, unroll_steps+1]
+            # [B, T, unroll_steps+1, ...], for the scalar case, the shape is
+            # [B, T, unroll_steps+1, 1]
             target = replay_buffer.get_field(self._target_fields, env_ids,
                                              positions)
+            # scalar target case
+            if target.dim() == 3:
+                target = target.unsqueeze(3)
```

Comment:
Instead of modifying the shape here, I think the right way is to change `SimpleDecoder` to handle shapes different from vector.

Reply:
Now handling it in decoder.